packages/android_alarm_manager/example/lib/main.dart

```diff
@@ -5,37 +5,79 @@
 import 'dart:async';
 import 'dart:isolate';
 
-import 'package:flutter/widgets.dart';
 import 'package:android_alarm_manager/android_alarm_manager.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter/widgets.dart';
+
+final FirebaseAuth firebaseAuth = FirebaseAuth.instance;
+FirebaseUser firebaseUser;
+
+Future<Null> ensureFirebaseUser() async {
+  if (firebaseUser == null) {
+    firebaseUser = await firebaseAuth.currentUser();
+    if (firebaseUser == null) {
+      firebaseUser = await firebaseAuth.signInAnonymously();
+    }
+  }
+}
+
+class HelloMessage {
+  final DateTime _now;
+  final String _msg;
+  final int _isolate;
+  final FirebaseUser _user;
+  final String _token;
+
+  HelloMessage(this._now, this._msg, this._isolate, this._user, this._token);
+
+  @override
+  String toString() {
+    return "[$_now] $_msg "
+        "isolate=$_isolate "
+        "user='$_user' "
+        "token=$_token";
+  }
+}
 
 void printHello() {
-  final DateTime now = new DateTime.now();
-  final int isolateId = Isolate.current.hashCode;
-  print("[$now] Hello, world! isolate=$isolateId function='$printHello'");
+  ensureFirebaseUser().then((_) {
+    firebaseUser.getIdToken().then((String idToken) {
+      print(new HelloMessage(new DateTime.now(), "Hello, world!",
+          Isolate.current.hashCode, firebaseUser, idToken));
```

Comment:
Trailing commas help `dartfmt` avoid 4-space indents.
In case you want to align to Flutter style guide.
Similarly a number of times below.

Reply:
Done.